src/server.c

```diff
@@ -3230,16 +3234,17 @@ void call(client *c, int flags) {
     dirty = server.dirty;
     incrCommandStatsOnError(NULL, 0);
 
+    const long long call_timer = ustime();
+
     /* Update cache time, in case we have nested calls we want to
      * update only on the first call*/
     if (server.fixed_time_expire++ == 0) {
-        updateCachedTime(0);
+        updateCachedTimeWithUs(0,call_timer);
     }
     server.in_nested_call++;
 
-    elapsedStart(&call_timer);
     c->cmd->proc(c);
-    const long duration = elapsedUs(call_timer);
+    const long duration = call_timer - ustime();
```

Comment:
should be the other way around.. that's why the tests are failing.
```suggestion
    const long duration = ustime() - call_timer;
```